src/typechecker.erl

```diff
@@ -3261,8 +3264,11 @@ type_check_cons_union(Env, [_ | Tys], H, T) ->
 
 get_bounded_fun_type_list(Name, Arity, Env, P) ->
     case maps:find({Name, Arity}, Env#env.fenv) of
-        {ok, Types} ->
-            typelib:remove_pos(Types);
+        %% TODO: https://github.com/josefs/Gradualizer/issues/388
+        {ok, Types} when is_list(Types) ->
+            [ typelib:remove_pos(Ty) || Ty <- Types ];
+        {ok, Type} ->
+            typelib:remove_pos(Type);
```

Comment:
The name of this function is unfortunate, since it doesn't always return a list of bounded fun types. Sometimes it just returns the type `any()`.


For functions without a spec, `any()` is set in `create_fenv/2`. We could replace this default type with a type on the form `fun((any(), ...., any()) -> any())` of the correct arity. (This is already done in `gradualizer_db:make_function_type/1` which could be moved/exported/reused.)